File: mobile/ripple_app/lib/models/api_response.dart
class ApiResponse<T> {
  final bool success;
  final String message;
  final T? data;

  ApiResponse({
    required this.success,
    required this.message,
    this.data,
  });

  bool get isSuccess => success;

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic json)? fromJsonT,
  ) {
    return ApiResponse<T>(
      success: json["success"] ?? false,
      message: json["message"] ?? 'Unknown error',
      data: json['data'] != null && fromJsonT != null
          ? fromJsonT(json['data'])
          : null,
    );
  }
}

File: backend/user-service/src/main/java/com/ripple/user_service/response/ApiResponse.java
package com.ripple.user_service.response;

import lombok.Getter;

@Getter
public class ApiResponse <T>{
    private boolean success;
    private String message;
    private T data;



    public ApiResponse(boolean success, String message, T data){
        this.success = success;
        this.message = message;
        this.data = data;
    }

    public static <T> ApiResponse success(String message, T data){
        return new ApiResponse(true, message, data);
    }
    public static <T> ApiResponse error(String message){
        return new ApiResponse(false, message, null);
    }
}
